dockstore-integration-testing/src/test/java/io/dockstore/webservice/metrics/MetricsDataS3ClientIT.java

```diff
@@ -0,0 +1,142 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.metrics;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.common.io.Files;
+import io.dockstore.webservice.core.metrics.MetricsData;
+import io.dockstore.webservice.core.metrics.MetricsDataS3Client;
+import io.dropwizard.testing.ResourceHelpers;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest;
+import software.amazon.awssdk.services.s3.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+public class MetricsDataS3ClientIT {
+    private static final String BUCKET_NAME = "dockstore.metrics.data";
+    private static URI localstackEndpoint;
+    private static S3Client s3Client;
+
+    @BeforeAll
+    public static void setup() throws URISyntaxException {
+        localstackEndpoint = new URI("http://localhost:4566");
+        s3Client = S3Client.builder().region(Region.US_EAST_1).endpointOverride(localstackEndpoint).build();
+        CreateBucketRequest request = CreateBucketRequest.builder().bucket(BUCKET_NAME).build();
+        s3Client.createBucket(request);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        // Delete all objects from the S3 bucket after each test
+        deleteBucketContents();
+    }
+
+    private static void deleteBucketContents() {
+        ListObjectsV2Request request = ListObjectsV2Request.builder().bucket(BUCKET_NAME).build();
+        ListObjectsV2Response response = s3Client.listObjectsV2(request);
+        List<S3Object> contents = response.contents();
+        contents.forEach(s3Object -> {
+            DeleteObjectRequest deleteObjectRequest = DeleteObjectRequest.builder().bucket(BUCKET_NAME).key(s3Object.key()).build();
+            s3Client.deleteObject(deleteObjectRequest);
+        });
+    }
+
+    @Test
+    void testS3Prototype() throws IOException {
```

Comment:
just to help us visualize, what does the bucket look like after this, can use something like 
https://pypi.org/project/s3-tree/ and just add as comment to PR